transactionsx/transaction.go

```diff
@@ -241,32 +226,29 @@ type TransactionMutableItemMeta struct {
 
 // TransactionGetResult represents the result of a Get or GetOptional operation.
 type TransactionGetResult struct {
-	agent          *Agent
+	agent          *gocbcorex.Agent
 	oboUser        string
 	scopeName      string
 	collectionName string
 	key            []byte
 
 	Meta  *TransactionMutableItemMeta
 	Value []byte
-	Cas   Cas
+	Cas   uint64
 }
 
-// TransactionGetCallback describes a callback for a completed Get or GetOptional operation.
-type TransactionGetCallback func(*TransactionGetResult, error)
-
 // Get will attempt to fetch a document, and fail the transaction if it does not exist.
-func (t *Transaction) Get(opts TransactionGetOptions, cb TransactionGetCallback) error {
+func (t *Transaction) Get(ctx context.Context, opts TransactionGetOptions) (*TransactionGetResult, error) {
```

Comment:
Should ctx belong to the transaction rather than individual ops? Although pretty sure it's an anti-pattern to store context so perhaps not...